mmeds/database.py

```diff
@@ -766,17 +767,47 @@ def __init__(self, metadata, path, owner, study_type, reads_type, study_name, te
         :testing: A boolean. Changes the connection parameters for testing.
         """
         warnings.simplefilter('ignore')
+        super().__init__()
+        debug_log('MetadataUploader created with params')
+        debug_log({
+            'subject_metadata': subject_metadata,
+            'specimen_metadata': specimen_metadata,
+            'owner': owner,
+            'study_type': study_type,
+            'reads_type': reads_type,
+            'study_name': study_name,
+            'temporary': temporary,
+            'data_files': data_files,
+            'public': public,
+            'testing': testing
+        })
 
-        self.path = Path(path) / 'database_files'
         self.IDs = defaultdict(dict)
         self.owner = owner
         self.testing = testing
         self.study_type = study_type
         self.reads_type = reads_type
-        self.metadata = metadata
+        self.subject_metadata = Path(subject_metadata)
+        self.specimen_metadata = Path(specimen_metadata)
         self.study_name = study_name
         self.temporary = temporary
         self.public = public
+        self.datafiles = data_files
+        self.created = datetime.now()
+
+        if access_code is None:
+            self.access_code = get_salt(50)
+        else:
+            self.access_code = access_code
+
+        count = 0
+        new_dir = fig.DATABASE_DIR / ('{}_{}_{}'.format(self.owner, self.study_name, count))
+        while new_dir.is_dir():
+            new_dir = fig.DATABASE_DIR / ('{}_{}_{}'.format(self.owner, self.study_name, count))
+            count += 1
+        new_dir.mkdir()
+
+        self.path = Path(new_dir) / 'database_files'
```

Comment:
does this work for concatenating the paths? I've been using Path.joinpath()

Reply:
Yeah, this is the best part of pathlib. It'll automatically handle the right delimiters '\' for windows, fixing double '//'